pandas/io/formats/style_render.py

```diff
@@ -2098,35 +2105,54 @@ def _translate(self, styler: StylerRenderer, d: dict):
-                self._pseudo_css(styler.uuid, name, i, j, str(self.tt_data.iloc[i, j]))
-                for i in range(len(self.tt_data.index))
-                for j in range(len(self.tt_data.columns))
-                if not (
-                    mask.iloc[i, j]
-                    or i in styler.hidden_rows
-                    or j in styler.hidden_columns
-                )
+        if not self.as_title_attribute:
+            name = self.class_name
+            self.table_styles = [
+                style
+                for sublist in [
+                    self._pseudo_css(
+                        styler.uuid, name, i, j, str(self.tt_data.iloc[i, j])
+                    )
+                    for i in range(len(self.tt_data.index))
+                    for j in range(len(self.tt_data.columns))
+                    if not (
+                        mask.iloc[i, j]
+                        or i in styler.hidden_rows
+                        or j in styler.hidden_columns
+                    )
+                ]
+                for style in sublist
             ]
-            for style in sublist
-        ]
 
-        if self.table_styles:
-            # add span class to every cell only if at least 1 non-empty tooltip
+            if self.table_styles:
+                # add span class to every cell only if at least 1 non-empty tooltip
+                for row in d["body"]:
+                    for item in row:
+                        if item["type"] == "td":
+                            item["display_value"] = (
+                                str(item["display_value"])
+                                + f'<span class="{self.class_name}"></span>'
+                            )
+                d["table_styles"].extend(self._class_styles)
+                d["table_styles"].extend(self.table_styles)
+        else:
+            id_regex = re.compile(r"row(\d+)_col(\d+)")
             for row in d["body"]:
                 for item in row:
                     if item["type"] == "td":
-                        item["display_value"] = (
-                            str(item["display_value"])
-                            + f'<span class="{self.class_name}"></span>'
-                        )
-            d["table_styles"].extend(self._class_styles)
-            d["table_styles"].extend(self.table_styles)
-
+                        td_id = item["id"]
+                        match = id_regex.match(td_id)
+                        if match is not None:
+                            i, j = (int(x) for x in match.groups())
+                            if (
+                                not mask.iloc[i, j]
+                                or i in styler.hidden_rows
+                                or j in styler.hidden_columns
+                            ):
+                                item[
+                                    "attributes"
+                                ] += f'title="{self.tt_data.iloc[i, j]}"'
```

Comment:
Will there be certain characters that this will not allow. E.g i can see if " is in the string this will mess up tye HTML completely

Reply:
Is it safer to start this with a space `+= f' title...`

Reply:
If there are already attributes then yes, adding the space will definitely be safer.

For your previous comment, yes that would cause issues. However, if we use single quotes then the same situation can still occur. Honestly I'm not sure what the correct solution here is.  From some brief googling, it seems double quotes is the standard for attributes.

This SO post, makes it seem like this is a common problem and you'd have to escape the quotes in the title string. 

https://stackoverflow.com/a/3752794/9063378

Reply:
So I added a check to see if `"` vs `'` is used in the title string but I don't like it much at all. There's nothing stopping that both are used. So I really think this is just a thing that the end user would have to handle on their end to properly escape their quotes.

Reply:
OK thats fine, add the space and leave it up to the user.